crates/oxc_linter/src/rules/unicorn/prefer_dom_node_text_content.rs

```diff
@@ -0,0 +1,128 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode, Fix};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error(
+    "eslint-plugin-unicorn(prefer-dom-node-text-content): Prefer `.textContent` over `.innerText`."
+)]
+#[diagnostic(severity(warning), help("Replace `.innerText` with `.textContent`."))]
+struct PreferDomNodeTextContentDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferDomNodeTextContent;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Enforces the use of `.textContent` over `.innerText` for DOM nodes.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// There are some advantages of using .textContent,
+    /// like performance and more predictable behavior when updating it.
```

Comment:
maybe consider expanding this based on the stack overflow post/blog post.

something  like

> [innerText](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/innerText) returns the visible text contained in a node, while [textContent](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent) returns the full text, this means that `innerText is much more performance-heavy as it requires layout information to return the result.
> 
> [innerText](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/innerText) is defined only for HTMLElement objects, while [textContent](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent) is defined for all Node objects.

Reply:
Thanks for your suggestion. I made some adjustments to the description.